c2/c2-client-bundle/c2-client-service/src/main/java/org/apache/nifi/c2/client/service/C2HeartbeatFactory.java

```diff
@@ -225,4 +234,27 @@ private File getConfDirectory() {
 
         return confDirectory;
     }
+
+    private String calculateManifestHash(List<Bundle> loadedBundles) {
```

Comment:
As more factors will come into play with the new C2 operation implementations, I think it would be better to separate this logic into a class.

Reply:
You are absolutely right. For now I would leave it as is and as soon as the next extension comes in we can extract the logic at that point. Are you ok with this approach?